lib/message/decrypt.js

```diff
@@ -1,34 +1,39 @@
-import { openpgp } from '../openpgp';
+import { isStream } from '@openpgp/web-stream-tools';
+import { AES_CFB } from '@openpgp/asmcrypto.js/dist_es8/aes/cfb';
+import { decrypt, readSignature } from 'openpgp';
 import { decodeUtf8Base64, binaryStringToArray, arrayToBinaryString } from '../utils';
 import { serverTime } from '../serverTime';
 import { getMessage, handleVerificationResult } from './utils';
 import processMIME from './processMIME';
 import { getEncMessageFromEmailPM, getEncRandomKeyFromEmailPM } from './compat';
 import { VERIFICATION_STATUS } from '../constants';
 
-export async function decryptMessage(options) {
+export async function decryptMessage({ encryptedSignature, armor = true, ...options }) {
     const { publicKeys = [] } = options;
     options.date = typeof options.date === 'undefined' ? serverTime() : options.date;
 
+    const format = armor ? 'armored' : 'object';
+    options.format = typeof options.format === 'undefined' ? format : options.format;
+
     try {
         // If encryptedSignature exists, decrypt and use it
-        if (options.encryptedSignature) {
-            const decryptedSignature = await openpgp.decrypt({
+        if (encryptedSignature) {
+            const decryptedSignature = await decrypt({
                 ...options,
-                message: options.encryptedSignature,
+                message: await getMessage(encryptedSignature),
```

Comment:
`encryptedSignature` should already be given parsed (same assumption as the input message)